### PR TITLE
Add support for case-insensitive contains

### LINKED ISSUE #917: Case insensitive string.Contains
The other string comparison operators can all be used with case insensitive search. `string.Contains` can't at the moment and the reason is partly that it doesn't take a culture parameter. But we should work around that.

### PR DESCRIPTION
**Pull request by nirinchev**
Resolves #917

### CODE REVIEW

File: Shared/Realm.Shared/Linq/RealmResultsVisitor.cs [hunk 1]
```diff
@@ -331,14 +333,23 @@ internal override Expression VisitMethodCall(MethodCallExpression m)
                 }
             }
 
-            if (m.Method.DeclaringType == typeof(string))
+            if (m.Method.DeclaringType == typeof(string) ||
+                m.Method.DeclaringType == typeof(StringExtensions))
```
Comment (nirinchev) [inline Shared/Realm.Shared/Linq/RealmResultsVisitor.cs]: Initially I had it in an `else if`, but felt a lot of code was duplicated that way so moved it inside with the other string methods.

I'm not sure what's the argument about using it less frequently?

**Comment by nirinchev** [comment]
Yes, that's unfortunate, but it's MS's fault for breaking the pattern :P TBH, I like the `a.ToUpper().Contains("xx".ToUpper())` even less. It just doesn't seem intuitive (not to me at least), so I'll definitely need to look up some sort of documentation on how to do case-insensitive contains with Realm.

If we're to redesign this approach, I would prefer to support `IndexOf("aa") >= 0`, but that would imply we also support `IndexOf("bb") >= 1`, so once again, dead end.

Long story short, I think the current approach is the "least bad" one and opens the door for introducing support for `like` (should be quite cheap now).
